source/extensions/tracers/opentelemetry/resource_detectors/resource_provider.cc

```diff
@@ -14,7 +14,7 @@ namespace OpenTelemetry {
 namespace {
 bool isEmptyResource(const Resource& resource) { return resource.attributes_.empty(); }
 
-Resource createInitialResource(const std::string& service_name) {
+Resource createInitialResource(absl::string_view service_name) {
```

Comment:
Could you change `Resource` to use `flat_hash_map` for `attributes_` and delete the manual string conversion code below for string views?